html/json/WS.js

```diff
@@ -278,8 +278,9 @@ var WS = {
 	},
 
 	_addToTrie: function(t, key, value) {
-		for (var i = 0; i < key.length; i++) {
-			var c = key.charAt(i);
+		var p = key.split(/[.(]/);
```

Comment:
You can still match the whole id. As a side effect you can also add a * after any . in the id and it will match anything after that.
See the new unit test for an example.